lms/djangoapps/course_api/blocks/tests/test_views.py

```diff
@@ -148,14 +150,73 @@ def assert_true_iff(self, expression, predicate):
         else:
             self.assertFalse(expression)
 
-    def test_not_authenticated(self):
+    def test_not_authenticated_non_public_course_with_other_username(self):
+        """
+        Verify behaviour when accessing course blocks of a non-public course for another user anonymously.
+        """
+        self.client.logout()
+        self.verify_response(403)
+
+    def test_not_authenticated_non_public_course_with_all_blocks(self):
+        """
+        Verify behaviour when accessing all course blocks of a non-public course anonymously.
+        """
+        self.client.logout()
+        self.query_params.pop('username')
+        self.query_params['all_blocks'] = True
+        self.verify_response(403)
+
+    def test_not_authenticated_non_public_course_with_blank_username(self):
+        """
+        Verify behaviour when accessing course blocks of a non-public course for anonymous user anonymously.
+        """
+        self.client.logout()
+        self.query_params['username'] = ''
+        self.verify_response(403)
+
+    @mock.patch("course_api.blocks.permissions.is_course_public", Mock(return_value=True))
+    def test_not_authenticated_public_course_with_other_username(self):
+        """
+        Verify behaviour when accessing course blocks of a public course for another user anonymously.
+        """
         self.client.logout()
-        self.verify_response(401)
```

Comment:
It looks like we will no longer have any test cases here that check for a `401 Unauthorized` response;
are there any use cases where this could/should arise?

If so, can we add a relevant test case?

Reply:
I can't think of any situation where it would rise up. Now that this API is supported for unauthenticated users, this error isn't being raised explicitly, and I'm not aware of any explicit situations. 

I guess certain requests that are currently 403 Unauthorized, could alternatively be 401 Unauthenticated when raised in case of an anonymous user. 

I'm also wondering if it makes sense to throttle this API now that it's available publicly?

Reply:
@ormsbee Thoughts on throttling? That's a little out of my wheelhouse, presently.